File: mall-manager-web/src/main/webapp/js/service/specificationOptionService.js
// service层,与数据库交互
app.service("specificationOptionService",function($http){
    //BASE crud
    this.query=function(pageNum,pageSize,condition){
        return $http.post("../specificationOption/query?pageNum="+pageNum+"&pageSize="+pageSize,condition);
    }

    this.save=function (specificationOption) {
        return $http.post("../specificationOption/save",specificationOption);
    }

    this.remove=function (idList) {
        return $http.get("../specificationOption/delete?idList="+idList);
    }

    this.queryById=function (id) {
        return $http.get("../specificationOption/queryById?id="+id);
    }
});

File: mall-sellergoods-interface/src/main/java/com/marshal/sellergoods/service/SpecificationOptionService.java
package com.marshal.sellergoods.service;
import com.marshal.pojo.TbSpecificationOption;
import com.marshal.util.ResponseData;

import java.util.List;

public interface SpecificationOptionService {
	public List<TbSpecificationOption> findAll();

	public ResponseData query(TbSpecificationOption condition, int pageNum, int pageSize);

	public TbSpecificationOption queryById(Long id);

	public List<TbSpecificationOption> queryBySpecId(Long id);

	public void save(TbSpecificationOption tbSpecificationOption);

	public void delete(Long[] ids);
}
